token/program/src/processor.rs

```diff
@@ -757,6 +758,27 @@ impl Processor {
         Ok(())
     }
 
+    /// Processes a [GetAccountDataSize](enum.TokenInstruction.html) instruction
+    pub fn process_get_account_data_size(accounts: &[AccountInfo]) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        // make sure the mint is passed in
+        let _mint_account_info = next_account_info(account_info_iter)?;
+        set_return_data(&Account::LEN.to_le_bytes());
+        Ok(())
+    }
+
+    /// Processes an [InitializeImmutableOwner](enum.TokenInstruction.html) instruction
+    pub fn process_initialize_immutable_owner(accounts: &[AccountInfo]) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let token_account_info = next_account_info(account_info_iter)?;
+        let account = Account::unpack_unchecked(&token_account_info.data.borrow())?;
+        if account.is_initialized() {
+            return Err(TokenError::AlreadyInUse.into());
+        }
+        msg!("Not doing anything, instruction support is only for associated-token-account compatibility");
```

Comment:
nit: let's take the opportunity to advertise token 2022 here, as this log message will show up every time a new ATA is created!
```suggestion
        msg!("Please upgrade to SPL Token 2022 for immutable owner support");
```